src/config/utils.ts

```diff
@@ -42,7 +43,10 @@ const getLayerByKey = (layerKey: layerKeys): LayerType => {
       return throwInvalidLayer();
     case 'impact':
       if (checkRequiredKeys(ImpactLayerProps, definition, true)) {
-        return definition;
+        const apiConfig = definition.api
+          ? (mapKeys(definition.api, (v, k) => camelCase(k)) as StatsApi)
```

Comment:
I'm still getting my ropes around this segment of the codebase, correct me if I'm wrong!

Since we generally want to convert every key in the config from `snake_case` to `camelCase`, would it be more worthwhile to make [getLayerByKey](https://github.com/oviohub/prism-frontend/blob/master/src/config/utils.ts#L20) convert keys at a deeper (recursive) level instead of doing it once here?

Reply:
We could definitely do that. Alternatively, we could add a helper function to both check the StatsApi object and camelcase the keys with a typeguard:

Something like this:

```typescript
function parseStatsApiConfig(maybeConfig: { [key: string]: any }): StatsApi {
  const config = mapKeys(maybeConfig, (v, k) => camelCase(k))
  // check all required keys here
  return config as StatsApi
} 
```